usaspending_api/references/v2/views/naics.py

```diff
@@ -0,0 +1,178 @@
+
+    def _parse_and_validate_request(self, requested_naics: str, request_data) -> dict:
+        naics_filter = request_data.get("filter")
+        data = {"code": requested_naics, "description": naics_filter}
+        models = [
+            {"key": "code", "name": "code", "type": "integer", "allow_nulls": True, "optional": True},
+            {
+                "key": "description",
+                "name": "description",
+                "type": "text",
+                "text_type": "search",
+                "default": None,
+                "optional": True,
+                "allow_nulls": True,
+            },
+        ]
+        validated = TinyShield(models).block(data)
+        return validated
+
+    def _fetch_children(self, naics_code, naics_filter=None) -> list:
+        results = []
+        if naics_filter:
+            naics = NAICS.objects.annotate(text_len=Length("code")).filter(
+                **naics_filter, code__startswith=naics_code, text_len=len(naics_code) + 2
+            )
+        else:
+            naics = NAICS.objects.annotate(text_len=Length("code")).filter(
+                code__startswith=naics_code, text_len=len(naics_code) + 2
+            )
+        for naic in naics:
+            if len(naic.code) < 6:
+                result = OrderedDict()
+                result["naics"] = naic.code
+                result["naics_description"] = naic.description
+                result["count"] = (
+                    NAICS.objects.annotate(text_len=Length("code"))
+                    .filter(code__startswith=naic.code, text_len=6)
+                    .count()
+                )
+            else:
+                result = OrderedDict()
+                result["naics"] = naic.code
+                result["naics_description"] = naic.description
+                result["count"] = 0
+            results.append(result)
+        return results
+
+    def _filter_search(self, naics_filter: dict) -> dict:
+        tier1_codes = set([])
+        tier2_codes = set([])
+        tier3_codes = set([])
+
+        tier3_naics = NAICS.objects.annotate(text_len=Length("code")).filter(**naics_filter, text_len=6)
+        for naic in tier3_naics:
+            tier3_codes.add(naic.code)
+            tier2_codes.add(naic.code[:4])
+            tier1_codes.add(naic.code[:2])
+        tier2_naics = NAICS.objects.annotate(text_len=Length("code")).filter(**naics_filter, text_len=4)
+
+        for naic in tier2_naics:
+            tier2_codes.add(naic.code)
+            tier1_codes.add(naic.code[:2])
+        extra_tier2_naics = NAICS.objects.annotate(text_len=Length("code")).filter(code__in=tier2_codes, text_len=4)
+        tier1_naics = NAICS.objects.annotate(text_len=Length("code")).filter(text_len=2, code__in=tier1_codes)
+
+        tier2 = set(list(tier2_naics))
+        ex2 = list(extra_tier2_naics)
+        for naic in ex2:
+            tier2.add(naic)
```

Comment:
There's GOT to be a more elegant way to do this. You can add_all or append a set or something, can't you?

Reply:
Can either use set operators (set |= other) or the `.update()` method
https://docs.python.org/3/library/stdtypes.html#set